src/file/UrlRe.re

```diff
@@ -1,2 +1,18 @@
+type t;
+
+[@bs.new] external make: string => t = "URL";
+[@bs.new] external makeWithBase: (string, string) => t = "URL";
+[@bs.send.pipe : t] external host: string = "";
+[@bs.send.pipe : t] external hash: string = "";
+[@bs.send.pipe : t] external hostname: string = "";
+[@bs.send.pipe : t] external href: string = "";
+[@bs.send.pipe : t] external origin: string = "";
+[@bs.send.pipe : t] external pathname: string = "";
+[@bs.send.pipe : t] external port: string = "";
+[@bs.send.pipe : t] external protocol: string = "";
+[@bs.send.pipe : t] external search: string = "";
```

Comment:
Most of these should have setters. All except `origin` and `searchParams` I think. Would you mind adding adding those? Otherwise I can just do it when I merge.

Reply:
Done! I used `bs.set` and not `bs.send pipe` though. Just thought it'd be slightly clearer this way.